cmake/CPM.cmake

```diff
@@ -505,3 +509,62 @@ function(cpm_is_git_tag_commit_hash GIT_TAG RESULT)
     endif()
   endif()
 endfunction()
+
+macro(prettify_cpm_add_package OUT_VAR IS_IN_COMMENT)
+    set(oneValueArgs
+      NAME
+      FORCE
+      VERSION
+      GIT_TAG
+      DOWNLOAD_ONLY
+      GITHUB_REPOSITORY
+      GITLAB_REPOSITORY
+      GIT_REPOSITORY
+      SOURCE_DIR
+      DOWNLOAD_COMMAND
+      FIND_PACKAGE_ARGUMENTS
+      NO_CACHE
+      GIT_SHALLOW
+    )
+      set(multiValueArgs
+        OPTIONS
+      )
+
+      cmake_parse_arguments(CPM_AGS "" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
+
+      foreach(oneArgName ${oneValueArgs})
+          if(DEFINED CPM_AGS_${oneArgName})
+              if(${IS_IN_COMMENT})
+                  string(APPEND ${OUT_VAR} "#")
+              endif()
+              if(${oneArgName} STREQUAL "SOURCE_DIR")
+                  string(REPLACE ${CMAKE_SOURCE_DIR} "\${CMAKE_SOURCE_DIR}" CPM_ARGS_${oneArgName} ${CPM_ARGS_${oneArgName}} )
+              endif()
+              string(APPEND ${OUT_VAR} "    ${oneArgName} ${CPM_ARGS_${oneArgName}}\n")
+          endif()
+      endforeach()
+      foreach(multiArgName ${multiValueArgs})
+          if(DEFINED CPM_AGS_${multiArgName})
+              if(${IS_IN_COMMENT})
+                  string(APPEND ${OUT_VAR} "#")
+              endif()
+              string(APPEND ${OUT_VAR} "    ${multiArgName}\n")
+              foreach(singleOption ${CPM_AGS_${multiArgName}})
+                  if(${IS_IN_COMMENT})
+                      string(APPEND ${OUT_VAR} "#")
+                  endif()
+                  string(APPEND ${OUT_VAR} "        \"${singleOption}\"\n")
+              endforeach()
+          endif()
+      endforeach()
+
+    if(NOT "${CPM_AGS_UNPARSED_ARGUMENTS}" STREQUAL "")
+        string(REPLACE ";" " " CPM_AGS_UNPARSED_ARGUMENTS "${CPM_AGS_UNPARSED_ARGUMENTS}")
```

Comment:
This is risky, e.g. what if there's an escaped `;` somewhere in the unparsed arguments? Imo it's better to not touch them.

Reply:
Will it work to add in the `CPMDeclarePackage` some argument with `;` ? 
I tough it would create error if done like this:
```
CPMDeclarePackage(Dependency
      NAME Dependency
      SOURCE_DIR ${CMAKE_CURRENT_LIST_DIR}/local_dependency/dependency
      UPDATE_DISCONNECTED;ON;TESTCUSTOMDATA;TRUE
    )
```

Reply:
Hm that is good question, I have no idea how CMake handles mixing spaces and semicolons. The safest might be to loop over `${CPM_AGS_UNPARSED_ARGUMENTS}` and append them one-by-one with a space.